app/services/foreman_rh_cloud/cloud_request_forwarder.rb

```diff
@@ -0,0 +1,75 @@
+require 'rest-client'
+
+module ForemanRhCloud
+  class CloudRequestForwarder
+    include ::ForemanRhCloud::CloudAuth
```

Comment:
I suppose we should use certificate auth here, similar to what is done in [report upload job](https://github.com/theforeman/foreman_rh_cloud/blob/master/lib/foreman_inventory_upload/async/upload_report_job.rb#L32). We have a plan to get rid of token auth altogether.

Reply:
Do we have any docs on how to do cert auth to cloud? Inspired by [inventory upload](https://github.com/theforeman/foreman_rh_cloud/blob/master/lib/foreman_inventory_upload/scripts/uploader.sh.erb#L39), I tried:

```ruby
    RestClient::Request.execute(
        method: req_method,
        url: cloud_url,
        verify_ssl: ForemanRhCloud.verify_ssl_method,
        payload: forward_payload,
        ssl_client_cert: OpenSSL::X509::Certificate.new(certs[:cert]),
        ssl_client_key: OpenSSL::PKey::RSA.new(certs[:key]),
        headers: {
          params: forward_params,
        }
```
which gives me 401: Authentication missing. @lphiri, could you point me in the right direction please?

Reply:
I managed to authenticate with certs to cert.cloud however I was getting 404 for endpoints that were previously working, so it seems we'll need to stick to token based auth for the time being:

```
api/v1/static/release/insights-core.egg?branch_id=20d6765d-b5aa-4bf2-b8c3-efda358dff98", "Accept"=>"*/*", "User-Agent"=>"rest-client/2.1.0 (linux-gnu x86_64) ruby/2.5.5p157"
 c06cca6e |
2020-10-20T10:28:40 [D|app|c06cca6e] # => 404 NotFound | text/plain 31 bytes, 0.30s

/api/v1/static/uploader.v2.json?branch_id=20d6765d-b5aa-4bf2-b8c3-efda358dff98", "Accept"=>"*/*", "User-Agent"=>"rest-client/2.1.0 (linux-gnu x86_64) ruby/2.5.5p157"
 f604e6d6 |
2020-10-20T10:28:41 [D|app|f604e6d6] # => 404 NotFound | text/plain 31 bytes, 0.22s
 f604e6d6 | 
```

Reply:
@xprazak2 i would not expect changing auth type to be the cause of the 404s.  Are you saying you get a  404 when you use cert.cloud.redhat.com (vs cloud.redhat.com)?

Reply:
Yes, exactly.

Reply:
As we discussed on the mail, we can use `cloud.redhat.com` for those endpoints. Even more so, they don't require any authentication at all.